packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts

```diff
@@ -138,4 +103,108 @@ export class TokenAuthorizer extends Authorizer implements IAuthorizer {
+   * otherwise a 401 Unauthorized is returned to the client.
+   *
+   * @default - no regex filter will be applied.
+   */
+  readonly validationRegex?: string;
+
+  /**
+   * The request header mapping expression for the bearer token. This is typically passed as part of the header, in which case
+   * this should be `method.request.header.Authorizer` where Authorizer is the header containing the bearer token.
+   * @see https://docs.aws.amazon.com/apigateway/api-reference/link-relation/authorizer-create/#identitySource
+   * @default 'method.request.header.Authorization'
+   */
+  readonly identitySource?: string;
+}
+
+/**
+ * Token based lambda authorizer that recognizes the caller's identity as a bearer token,
+ * such as a JSON Web Token (JWT) or an OAuth token.
+ * Based on the token, authorization is performed by a lambda function.
+ *
+ * @resource AWS::ApiGateway::Authorizer
+ */
+export class TokenAuthorizer extends LambdaAuthorizer {
+
+  constructor(scope: Construct, id: string, props: TokenAuthorizerProps) {
+    super(scope, id, props);
+  }
+
+  protected createResource(props: TokenAuthorizerProps): CfnAuthorizer {
+    const restApiId = Lazy.stringValue({ produce: () => this.restApiId });
+
+    return new CfnAuthorizer(this, 'Resource', {
+      name: props.authorizerName ?? this.node.uniqueId,
+      restApiId,
+      type: 'TOKEN',
+      authorizerUri: `arn:aws:apigateway:${Stack.of(this).region}:lambda:path/2015-03-31/functions/${props.handler.functionArn}/invocations`,
+      authorizerCredentials: props.assumeRole?.roleArn,
+      authorizerResultTtlInSeconds: props.resultsCacheTtl?.toSeconds() ?? 0,
+      identitySource: props.identitySource || 'method.request.header.Authorization',
+      identityValidationExpression: props.validationRegex,
+    });
+  }
+}
+
+/**
+ * Properties for RequestAuthorizerProps
+ */
+export interface RequestAuthorizerProps extends LambdaAuthorizerProps {
+  /**
+   * An array of request header mapping expressions for identities. Supported parameter types are
+   * Header, Query String, Stage Variable, and Context. For instance, extracting an authorization
+   * token from a header would use the identity source `method.request.header.Authorizer`.
+   *
+   * Note: API Gateway uses the specified identity sources as the request authorizer caching key. When caching is
+   * enabled, API Gateway calls the authorizer's Lambda function only after successfully verifying that all the
+   * specified identity sources are present at runtime. If a specified identify source is missing, null, or empty,
+   * API Gateway returns a 401 Unauthorized response without calling the authorizer Lambda function.
+   *
+   * @see https://docs.aws.amazon.com/apigateway/api-reference/link-relation/authorizer-create/#identitySource
+   * @default no identity sources
+   */
+  readonly identitySource?: string[];
```

Comment:
Idea: would be nice to have supporting class/methods that can help the user generate these values. Something like,  `IdentitySource.header('Authorization')`, `IdentitySource.context('https')`, etc.

Non-blocking for this PR, since we can always supplement this later.

Reply:
I **love** this idea, and I'm adding it!